core/services/functions/listener.go

```diff
@@ -386,40 +408,44 @@ func (l *FunctionsListener) getMaxSecretsSize(flags RequestFlags) uint32 {
 	return l.pluginConfig.MaxSecretsSizesList[idx]
 }
 
-// TODO (FUN-662): call from LogPoller in a separate goroutine
-func (l *FunctionsListener) HandleOracleRequestV1(request *functions_coordinator.FunctionsCoordinatorOracleRequest, lb log.Broadcast) {
+func (l *FunctionsListener) handleOracleRequestV1(request evmrelayTypes.OracleRequest) {
 	defer l.shutdownWaitGroup.Done()
-	l.logger.Infow("oracle request v1 received", "requestID", formatRequestId(request.RequestId))
+	var requestId [32]byte
+	copy(requestId[:], request.RequestId)
+	l.logger.Infow("oracle request v1 received", "requestID", formatRequestId(requestId))
 	ctx, cancel := l.getNewHandlerContext()
 	defer cancel()
 
 	callbackGasLimit := uint32(request.CallbackGasLimit)
 	newReq := &Request{
-		RequestID:                  request.RequestId,
-		RequestTxHash:              &request.Raw.TxHash,
+		RequestID:                  requestId,
+		RequestTxHash:              (*common.Hash)(request.TxHash),
 		ReceivedAt:                 time.Now(),
 		Flags:                      request.Flags[:],
 		CallbackGasLimit:           &callbackGasLimit,
-		CoordinatorContractAddress: &request.Raw.Address,
+		CoordinatorContractAddress: (*common.Address)(request.OracleContract),
 	}
 	if err := l.pluginORM.CreateRequest(newReq, pg.WithParentCtx(ctx)); err != nil {
 		if errors.Is(err, ErrDuplicateRequestID) {
-			l.logger.Warnw("received a log with duplicate request ID", "requestID", formatRequestId(request.RequestId), "err", err)
-			l.markLogConsumed(lb, pg.WithParentCtx(ctx))
+			l.logger.Warnw("received a log with duplicate request ID", "requestID", formatRequestId(requestId), "err", err)
+			// l.markLogConsumed(lb, pg.WithParentCtx(ctx)) // TODO?
```

Comment:
@bolekk how do we want to handle consuming logs?

Maybe LogPollerWrapper marks them consumed when they are queried?
Otherwise an API method to mark them consumed?

Reply:
I think LogPollerWrapper will guarantee that every log is delivered once. I will confirm soon.